Src/ElasticScale.Client/ShardManagement/Cache/PerfCounterInstance.cs

```diff
@@ -107,56 +107,79 @@ public PerfCounterInstance(string shardMapName)
+                            PerformanceCounters.ShardManagementPerformanceCounterCategory);
                     }
+                    else
+                    {
+                        // Check if specific instance exists
+                        if (PerformanceCounterCategory.InstanceExists(_instanceName,
+                            PerformanceCounters.ShardManagementPerformanceCounterCategory))
+                        {
+                            // As performance counters are created with Process lifetime and instance name is unique (PID + shard map name),
+                            // this should never happen. Trace out error and silently continue.
+                            Tracer.TraceWarning(TraceSourceConstants.ComponentNames.PerfCounter,
+                                "create",
+                                "Performance counter instance {0} already exists, no performance data will be collected.",
+                                _instanceName);
+                        }
+                        else
+                        {
+                            // now initialize all counters for this instance
+                            _counters = new Dictionary<PerformanceCounterName, PerformanceCounterWrapper>();
+
+                            foreach (PerfCounterCreationData d in PerfCounterInstance.counterList)
+                            {
+                                _counters.Add(d.CounterName,
+                                    new PerformanceCounterWrapper(
+                                        PerformanceCounters.ShardManagementPerformanceCounterCategory, _instanceName,
+                                        d.CounterDisplayName));
+                            }
 
-                    // check that atleast one performance counter was created, so that we can remove instance as part of Dispose()
-                    _initialized = _counters.Any(c => c.Value._isValid = true);
+                            // check that atleast one performance counter was created, so that we can remove instance as part of Dispose()
+                            _initialized = _counters.Any(c => c.Value._isValid = true);
+                        }
+                    }
                 }
             }
+            catch (Exception e)
+            {
+                // Note: If any of the initialization calls throws, log the exception and silently continue.
+                // No perf data will be collected in this case.
+                // All other non-static code paths access PerformanceCounter and PerformanceCounterCategory
+                // objects only if _initialized is set to true.
+
+                Tracer.TraceWarning(TraceSourceConstants.ComponentNames.PerfCounter,
+                    "PerfCounterInstance..ctor",
+                    "Exception caught while creating performance counter instance, no performance data will be collected. Exception: {0}",
+                    e.Message);
```

Comment:
I'd prefer if you just wrote the entire exception `e.ToString()` so that stack and inner exceptions aren't lost.